_posts/2020-06-18-Metal3-dev-env-BareMetal-Cluster-Deployment.md

```diff
@@ -0,0 +1,396 @@
+
+| Parameter           | Description                  | Default                  |
+| ------------------- | ---------------------------- | ------------------------ |
+| CAPI_VERSION        | Version of Metal3 API4       | v1alpha3        |
+| POD_CIDR            | Pod Network CIDR             | 192.168.0.0/18           |
+| CLUSTER_NAME        | Name of bare metal cluster   | test1                    |
+
+<br/>
+
+
+### Steps Involved
+
+All the scripts for cluster provisioning or deprovisioning are located at - [`${metal3-dev-env}/scripts/v1alphaX/`](https://github.com/metal3-io/metal3-dev-env/tree/master/scripts/v1alphaX). The scripts call a common playbook which handles all the tasks that are available.
+
+> info "Note"
+> Here is a depiction of the common steps, mainly involving generating templates and config files. 
+
+<br/>
+
+!["A diagram depicting the Generate Templates workflow"](/assets/2020-06-18-Metal3-dev-env-BareMetal-Cluster-Deployment/metal3-generate-templates.png)
+<br/>
+<br/>
+
+The steps involved in the process are :
+
+- The script calls an ansible playbook with necessary parameter ( from env variables and defaults )
+- The playbook executes the role -, [`${metal3-dev-env}/vm-setup/roles/v1aX_integration_test`](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test), which runs the corresponding [task_file](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test/tasks) for provisioning/deprovisioning the cluster/controlplane or a worker
+- There are [templates](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test/templates) in the role, which are copied to the `manifest` directory. They are used for generating configuration for the cluster and kubeadm, which is then supplied to the Kubernetes module of ansible to create the cluster.
+- During provisioning, first the `clusterctl` env file is generated, then and `manifest` file is created/updated using the `clusterctl` tool, which is then supplied to `kustomize` tool to create the config files for provisioning the resource, from the available templates.
+- The manifests are generated for - cluster, controlplane and machinedeployment, and the worker manifests are merged with the deployment manifests.
+- Finally a yaml file is generated with all the details for the cluster/controlplane/worker, which is then passed to K8s module in ansible which creates the resources.
+- There are also manifests stored in ${Manifests} directory - [`${metal3-dev-env}/vm-setup/roles/v1aX_integration_test/files/manifests`](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test), for the cluster which can be used to deprovision the cluster at a later point in time.
```

Comment:
we always regenerate the target pieces (in /tmp) even before deprovisioning. It might be more straightforward to use those files (in /tmp) for deprovisioning

Reply:
I have changed this section quite a bit, let me know if the new content makes sense